test/helpers/eip712.js

```diff
@@ -36,10 +36,40 @@ function hashTypedData(domain, structHash) {
   );
 }
 
+function hashNestedTypedDataType(contentsTypeName, contentsType) {
+  return ethers.solidityPackedKeccak256(
+    ['string'],
+    [
+      `TypedDataSign(${contentsTypeName}bytes1 fields,string name,string version,uint256 chainId,address verifyingContract,bytes32 salt,uint256[] extensions)${contentsType}`,
```

Comment:
THis string is 
```TypedDataSign(SomeTypebytes1 fields,string name,string version,uint256 chainId,address verifyingContract,bytes32 salt,uint256[] extensions)SomeType()```

IMO it sould be
```
TypedDataSign(SomeType something,bytes1 fields,string name,string version,uint256 chainId,address verifyingContract,bytes32 salt,uint256[] extensions)SomeType()```

Reply:
That is documented properly in the reference implemenation https://ethereum-magicians.org/t/erc-7739-readable-typed-signatures-for-smart-accounts/20513#reference-implementation-23

But not in this section https://ethereum-magicians.org/t/erc-7739-readable-typed-signatures-for-smart-accounts/20513#typeddatasign-final-hash-11

Reply:
The more I read the tests, the worst I feel about it all. This should have been catched by some test doing a full `ethers.TypedDataEncoder.hash` or an actual `signTypedData` on the full EIP712 wrapper, but we don't have that. Instead we have some tests taht are overly complexe for testing actually nothing ...